tide-whisperer.go

```diff
@@ -281,6 +290,18 @@ func main() {
 		iter := storage.GetDeviceData(queryParams)
 		defer iter.Close()
 
+		var parametersHistory map[string]interface{}
+		var parametersHistoryErr error
+		if inArray("pumpSettings", queryParams.Types) || len(queryParams.Types) == 1 {
```

Comment:
What is the point of this test?

Reply:
We only request the history of parameters  when the type is "pumpSettings" or when no type is requested. 

The problem is that  queryParams.Types is returned by strings.Split function. With an empty string split returns an array : [""] which has a length of 1. 

The test should in reality be this one : 
```suggestion
		if inArray("pumpSettings", queryParams.Types) || (len(queryParams.Types) == 1 && queryParams.Types[0]=="")  {
```